torch/nn/parameter.py

```diff
@@ -24,6 +24,11 @@ def __new__(cls, data=None, requires_grad=True):
             data = torch.Tensor()
         return torch.Tensor._make_subclass(cls, data, requires_grad)
 
+    def __deepcopy__(self, memo):
+        result = type(self)(self.data.clone(), self.requires_grad)
```

Comment:
Can't we do this conditionally?
```suggestion
        if id(self) not in memo:
            # do this
        else:
            return memo[id(self)]
```

Reply:
Thanks! Yeah, I forgot to do that.